### PR TITLE
refactor: Cleanup DHC Service

### PR DESCRIPTION
**Pull request by bmingles**
- Refactored DhcService to be more consistent with patterns used by DheService
- Psks are now stored in secret storage instead of keeping credentials in memory cache. They now behave similar to private keys in that one that has been saved will be automatically used when connecting to the server.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -219,7 +219,11 @@
         "icon": "$(refresh)"
       },
       {
-        "command": "vscode-deephaven.createAuthenticatedClient",
+        "command": "vscode-deephaven.createCoreAuthenticatedClient",
+        "title": "Create Authenticated Client"
+      },
+      {
+        "command": "vscode-deephaven.createDHEAuthenticatedClient",
         "title": "Create Authenticated Client"
```
[the remark this answers is not in the document]
Comment (bmingles) [reply package.json]: Technically, the titles never show in the UI, but I've update for clarity.

File: src/services/cache/CoreJsApiCache.ts [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+/**
+ * Cache Core jsapi instances by URL.
+ */
+export class CoreJsApiCache extends ByURLAsyncCache<typeof DhcType> {
+  constructor() {
+    super(async url =>
+      initDhcApi(url, getTempDir({ subDirectory: urlToDirectoryName(url) }))
+    );
+  }
+}
```
[the remark this answers is not in the document]
Comment (bmingles) [reply src/services/cache/CoreJsApiCache.ts:15]: @mofojed Workers store the respective api by the worker URL. Am I misunderstanding?

[the remark this answers is not in the document]
Comment (mofojed) [reply src/services/cache/CoreJsApiCache.ts:15]: @bmingles Workers each have their own JS API URL, correct. However, in the case of DHE, you could have multiple workers that have the same worker kind configuration/engine; in which case you're just downloading the same JS API multiple times from different URLs. We optimize this in DHE by memoizing fetching the API on the worker kind/engine type: https://github.com/deephaven-ent/iris/blob/0fccc395d44974571e71085c762b68c151c7ca58/web/client-ui/src/main/AppMainContainer.tsx#L587

[the remark this answers is not in the document]
Comment (bmingles) [reply src/services/cache/CoreJsApiCache.ts:15]: oh gotcha. Mind if I handle that in a separate PR to keep things cleaner? That's technically not a new concept introduced by this one.

[the remark this answers is not in the document]
Comment (bmingles) [reply src/services/cache/CoreJsApiCache.ts:15]: @mofojed I've created [Cache DHE Core+ jsapis by worker kind / engine type #174](https://github.com/deephaven/vscode-deephaven/issues/174) to follow up on this.

**Review by mofojed — APPROVED** [review]

**Pull request merged**